debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -449,20 +451,34 @@ private Scn calculateEndScn(OracleConnection connection, Scn startScn, Scn prevE
             if (!topMiningScnInFarFuture) {
                 streamingMetrics.changeSleepingTime(true);
             }
-            LOGGER.debug("Using current SCN {} as end SCN.", currentScn);
+            LOGGER.debug("Using current SCN {} as end SCN.", currentScn.getFullString());
             return currentScn;
         }
         else {
             if (prevEndScn != null && topScnToMine.compareTo(prevEndScn) <= 0) {
-                LOGGER.debug("Max batch size too small, using current SCN {} as end SCN.", currentScn);
+                LOGGER.debug("Max batch size too small, using current SCN {} as end SCN.", currentScn.getFullString());
                 return currentScn;
             }
             streamingMetrics.changeSleepingTime(false);
             if (topScnToMine.compareTo(startScn) < 0) {
-                LOGGER.debug("Top SCN calculation resulted in end before start SCN, using current SCN {} as end SCN.", currentScn);
+                LOGGER.debug("Top SCN calculation resulted in end before start SCN, using current SCN {} as end SCN.", currentScn.getFullString());
                 return currentScn;
             }
-            LOGGER.debug("Using Top SCN calculation {} as end SCN.", topScnToMine);
+
+            if (prevEndScn != null) {
+                if (prevEndScn.getTime() != null) {
+                    long seconds = ChronoUnit.SECONDS.between(prevEndScn.getTime(), currentScn.getTime());
+                    if (currentScn.subtract(prevEndScn).longValue() > 10000000 && seconds < 20) {
```

Comment:
Could we extract the SCN gap value, `10000000`, and time variance value, `20`, to configuration options and set these as defaults to those options allowing users to adjust these for their environment?